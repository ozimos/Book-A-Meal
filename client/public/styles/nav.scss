nav.flexbox{
    background-color: #fafafa;
    a{
        font-family: Roboto, Monserrat, 'Segoe UI', Tahoma, Verdana, sans-serif;
        color: #2f80ed; 
        line-height: normal;                    
        &.active{
            font-weight: bold;
          } 
      }
  }
