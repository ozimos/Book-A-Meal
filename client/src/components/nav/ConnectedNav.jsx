import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import '../../../public/styles/nav.scss';

const Nav = ({ user, authenticated }) =>
  (
    <div className="navbar-fixed">
      <nav className="flexbox">
        <h3 className="shrink heading">
          <Link to="/">
        Book A Meal
          </Link>
        </h3>
        <div className="flexbox">
          {user.isCaterer &&
          <NavLink
            activeClassName="active"
            className="nav-item long_string"
            to="/dashboard"
          >
        DashBoard
          </NavLink >}
          {user.isCaterer &&
          <NavLink
            activeClassName="active"
            className="nav-item"
            to="/meals"
          >
          Meals
          </NavLink >}
          {user.isCaterer &&
          <NavLink
            activeClassName="active"
            className="nav-item"
            to="/menu"
          >
          Menu
          </NavLink>}
          {!user.isCaterer &&
          <NavLink
            activeClassName="active"
            className="nav-item"
            to="/orders"
          >
          Meal Booking
          </NavLink>}
          {authenticated &&
          <Link to="/login">
          Log Out
          </Link>}
          {!authenticated &&
          <Link to="/login">
          Log In
          </Link>}
        </div>
      </nav>
    </div>
  );
Nav.defaultProps = {
  user: { isCaterer: false }
};
Nav.propTypes = {
  user: PropTypes.shape({
    isCaterer: PropTypes.bool,
  }),
  authenticated: PropTypes.bool.isRequired,
};
const mapStateToProps = state => ({
  user: state.loginReducer.user.data,
  authenticated: state.loginReducer.authenticated,
});
export { Nav };
export default connect(mapStateToProps)(Nav);
