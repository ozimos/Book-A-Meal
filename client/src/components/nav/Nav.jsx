import React from 'react';
import { Link, NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import '../../../public/styles/nav.scss';

const Nav = ({ user }) =>
  (
    <div className="navbar-fixed">
      <nav className="flexbox">
        <h3 className="shrink heading">Book A Meal</h3>
        <div className="flexbox">
          {user.isCaterer &&
          <NavLink activeClassName="active" className="nav-item long_string" to="/dashboard">
        DashBoard
          </NavLink >}
          {user.isCaterer &&
          <NavLink activeClassName="active" className="nav-item" to="/meals">
          Meals
          </NavLink >}
          {user.isCaterer &&
          <NavLink activeClassName="active" className="nav-item" to="/menu">
          Menu
          </NavLink>}
          <NavLink activeClassName="active" className="nav-item" to="/orders">
          Meal Booking
          </NavLink>
          <Link to="/login">
          Log Out
          </Link>
        </div>
      </nav>
    </div>
  );
Nav.propTypes = {
  user: PropTypes.shape({
    isCaterer: PropTypes.bool,
  }).isRequired,
};
const mapStateToProps = state => ({
  user: state.loginReducer.user.data,
  authenticated: state.loginReducer.authenticated,
});
export { Nav };
export default connect(mapStateToProps)(Nav);
