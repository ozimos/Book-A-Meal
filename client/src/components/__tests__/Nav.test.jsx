/* global React:false, shallow:false toJson:false */

import { Nav } from '../nav/Nav';

describe('Nav Component', () => {
  const user = { isCaterer: true };

  it('should render without throwing an error', () => {
    expect(shallow(<Nav user={user} />)
      .exists(<nav className="flexbox" />)).toBe(true);
  });
  it('renders correctly', () => {
    const wrapper = shallow(<Nav user={user} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

