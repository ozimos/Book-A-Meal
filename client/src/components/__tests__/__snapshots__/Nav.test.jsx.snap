// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav Component renders correctly 1`] = `
<div
  className="navbar-fixed"
>
  <nav
    className="flexbox"
  >
    <h3
      className="shrink heading"
    >
      Book A Meal
    </h3>
    <div
      className="flexbox"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="nav-item long_string"
        to="/dashboard"
      >
        DashBoard
      </NavLink>
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="nav-item"
        to="/meals"
      >
        Meals
      </NavLink>
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="nav-item"
        to="/menu"
      >
        Menu
      </NavLink>
      <Link
        replace={false}
        to="/login"
      >
        Log In
      </Link>
    </div>
  </nav>
</div>
`;
